include/bl_stash.rst

```diff
@@ -0,0 +1,44 @@
+.. _doc_bl_stash:
+
+Bootloader provisioning data (bl_stash)
+#######################################
+
+The bootloader provision data library is used by the :ref:`bootloader` to read from and write to the provisioning data associated with it.
+
+The library can also be included into the app if it needs to read provisioning data.
+The data will be placed in the 'OTP' region of UICR if available.
+Otherwise, the data is placed in a regular flash page, in which case the bootloader will block write access to it before booting the next image.
+When the data is blocked from writes, some functions do not work when they are called from the app.
+The provisioning data cannot be read from non-secure apps.
+
+The provision data contains:
+ - Image slot sizes
+ - Public keys
+ - Invalidation tokens used to revoke public keys
+ - Monotonic version counter
+
+See :ref:`bootloader_provisioning` for more information about the provisioning data and how the bootloader uses it.
+
+There are tests for the library at :file:`tests/subsys/bootloader/bl_stash/`.
+
+Monotonic version counter
+*************************
+
+The provision page can contain a monotonic version counter for the app.
+This is implemented as a series of 16 bit values, where each update to the counter is written to the next available slot.
+When reading the counter, bl_provision iterates over each slot checking that each is larger than the previous, to guarantee that it is monotonic (always increases).
```

Comment:
Need decision point on strictly monotonic vs monotonic accepting any-value-higher-than
The decision needs to be documented to ensure clarity on what is meant by "monotonic"

Reply:
Added as a discussion point on #1917